### PR TITLE
feature: application group improvements

### PR DESCRIPTION
**Pull request by radko93**
This PRs adds cleanup for observers and additionally avoids observers being notified twice with the same result.

### CODE REVIEW

File: app/assets/javascripts/ui_models/application_group.ts [hunk 1]
```diff
@@ -121,6 +123,10 @@ export class ApplicationGroup {
     if (this.application) {
       callback();
     }
+
+    return () => {
+      pull(this.changeObservers, callback)
```
Comment (baptiste-grob) [inline app/assets/javascripts/ui_models/application_group.ts:128]: Looking at the lodash source, using `pull` seems overkill when we only need to check for referential equality. I think a combination of `indexOf` and `splice` would be more efficient, maybe as a utility function? (see `utils.ts`)

Comment (radko93) [reply app/assets/javascripts/ui_models/application_group.ts:128]: Updated.

Comment (mobitar) [reply app/assets/javascripts/ui_models/application_group.ts:128]: SNJS has slowly been in the process of removing dependence on lodash in favor of utils. So we have [`removeFromArray`](https://github.com/standardnotes/snjs/blob/004/lib/utils.ts#L175) that can be imported from SNJS.

File: app/assets/javascripts/ui_models/application_group.ts [hunk 2]
```diff
@@ -125,7 +124,8 @@ export class ApplicationGroup {
     }
 
     return () => {
-      pull(this.changeObservers, callback)
+      const indexOfObserver = this.changeObservers.indexOf(callback);
+      this.changeObservers.splice(indexOfObserver, 1);
```
Comment (mobitar) [inline app/assets/javascripts/ui_models/application_group.ts:128]: Should use `removeFromArray`. We use this 14 times already in the web code. No need to use splice directly.

File: app/assets/javascripts/ui_models/application_group.ts (hunk 1 above)
Comment (radko93) [reply app/assets/javascripts/ui_models/application_group.ts]: Thanks for the info. I will remove lodash in mobile code completely in 004.

Comment (mobitar) [reply app/assets/javascripts/ui_models/application_group.ts]: We don't have to totally phase out lodash if it's difficult. Just sort of a gradual process is good enough. But in most cases I believe snjs's utils collection offers a good replacement for most of the basic lodash utils.

**Pull request merged**